.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,61 @@
+# For most projects, this workflow file will not need changing; you simply need
+# to commit it to your repository.
+#
+# You may wish to alter this file to override the set of languages analyzed,
+# or to provide custom queries or build logic.
+#
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+# ******** NOTE ********
+
+name: "CodeQL"
+
+on:
+  pull_request:
+    branches:
+      - master
+      - opendistro-*
+  push:
+    branches:
+      - master
+      - opendistro-*
+  schedule:
+    - cron: '0 */3 * * *'
```

Comment:
why do we have a schedule if we are running on pull request only?

Reply:
This is to check if the CodeQL can give the same functionality in scanning repo as in WhiteSource. This would also help us to compare the vulnerability reports.